File: sa31/2024-11-06/project/JavaLesson1/src/local/hal/sa31/java/lesson1/ex4/CalcCircleAndBall.php
<?php
/**
 * SA31 Java文法 Lesson1 Ex4
 *
 * クラスファイル
 *
 * @author Shinzo SAITO
 */
class CalcCircleAndBall {
    /**
     * @var float 半径を表すプロパティ
     */
    private float $radius;

    /**
     * @var float 円周率を表す定数プロパティ
     */
    const PI = 3.14;

    /**
     * コンストラクタ
     * @param float $radius 半径
     */
    public function __construct(float $radius) {
        $this->radius = $radius;
    }

    /**
     * 半径を得るゲッタ
     * @return float 半径
     */
    public function getRadius(): float {
        return $this->radius;
    }

    /**
     * 円周を得るメソッド
     * @return float 計算された円周
     */
    public function getCircle(): float {
        $circle = 2 * self::PI * $this->radius;
        return $circle;
    }

    /**
     * 円の面積を得るメソッド
     * @return float 計算された円の面積
     */
    public function getArea(): float {
        $area = 2 * self::PI * $this->radius * $this->radius;
        return $area;
    }

    /**
     * 円の表面積を得るメソッド
     * @return float 計算された球の表面積
     */
    public function getSurface(): float {
        $surface = 4 * self::PI * $this->radius * $this->radius;
        return $surface;
    }

    /**
     * 円の体積を得るメソッド
     * @return float 計算された球の体積
     */
    public function getVolume(): float {
        $volume = 4 * self::PI * $this->radius * $this->radius * $this->radius / 3;
        return $volume;
    }
}

File: sa31/2024-11-06/project/JavaLesson1/src/local/hal/sa31/java/lesson1/ex4/CalcCircleAndBall.java
package local.hal.sa31.java.lesson1.ex4;

/**
 * SA31 Java文法 Lesson1 Ex4
 *
 * 円球計算クラス
 *
 * @author Shinzo SAITO
 */
public class CalcCircleAndBall {
    /**
     * 半径を表すフィールド
     */
    private double _radius = 0;
    /**
     * 円周率を表す定数フィールド
     */
    private static final double PI = 3.14;

    /**
     * コンストラクタ
     * @param radius 半径
     */
    public CalcCircleAndBall(double radius) {
        _radius = radius;
    }

    /**
     * 半径を得るメソッド
     * @return 半径
     */
    public double getRadius() {
        return _radius;
    }

    /**
     * 円周を得るメソッド
     * @return 計算された円周
     */
    public double getCircle() {
        return 2 * PI * _radius;
    }

    /**
     * 円の面積を得るメソッド
     * @return 計算された円の面積
     */
    public double getArea() {
        return PI * _radius * _radius;
    }

    /**
     * 球の表面積を得るメソッド
     * @return 計算された球の表面積
     */
    public double getSurface() {
        return 4 * PI * _radius * _radius;
    }

    /**
     * 球の体積を得るメソッド
     * @return 計算された球の体積
     */
    public double getVolume() {
        return 4 * PI * _radius * _radius * _radius / 3;
    }
}
